### ISSUE TITLE
S930 could not raise errors on noop function patterns usage

### ISSUE THREAD

**Issue by AMorgaut** [SonarSource/SonarJS#887]
**I want to provide the following feedback.**

**Rule key:** S930

It's not really a bug 
This rule raise what should be considered as false positives

**Example**

Consider this code

```javascript
const noop = () => {};

function foo(arg1, callback = noop) {
    const data = {...arg1};
    delete data.bar;
    callback(JSON.stringify(data));
    return data;
}
```

**Logs**
```
callback declared at line 2 expects 0 arguments but 1 was provided
```

**Expected**

Maybe This rule could ignore noop function patterns such as:

- const noop = () => {}
- const noop = function() {}
- const noop = Function.prototype

**Work Around**

We can add 'args' to the noop function and add a NOSONAR comment to not receive an error as "args" is not used...

```javascript
const noop = (...args) => {}; //NOSONAR
```

But it means that:
- there is some NOSONAR comment which is always suspicious
- the function is less a 'noop' function (may impact runtime perf)
